packages/client/src/generation/TSClient/Input.ts

```diff
@@ -141,13 +155,20 @@ ${indent(
     .map((arg) => {
       // This disables enumerable on JsonFilter path argument
       const noEnumerable = type.name.includes('Json') && type.name.includes('Filter') && arg.name === 'path'
-      return new InputField(arg, false, noEnumerable).toTS()
+      return new InputField(arg, false, noEnumerable, source).toTS()
     })
     .join('\n'),
   TAB_SIZE,
 )}
 }`
     return `
-export type ${type.name} = ${body}`
+export type ${this.getTypeName()} = ${body}`
+  }
+
+  private getTypeName() {
+    if (inputTypeNeedsGenericModelArg(this.type)) {
```

Comment:
This method looks exclusive to field references, should we rename it to make it more explixit?